src/utils/intentionHandlers/AssignDeposit.ts

```diff
@@ -43,6 +60,42 @@ export async function handleAssignDeposit(params: {
 
 	await context.validateAssignDepositStructure(intention)
 
+	// Determine submitter vault for nonce tracking
+	// 1. If inputs have `from` field, use that (all inputs must have the same `from` value per validator)
+	// 2. If no `from` field, determine from controller by querying vaults
```

Comment:
See note above about how we will never have a `from` vault for assigning deposits, since unassigned deposits by definition are not yet in a nonce, and will also therefore never have a vault nonce.